src/context.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+import * as React from 'react';
+import {createPlugin} from 'fusion-core';
+import type FusionApp, {FusionPlugin, Middleware, Token} from 'fusion-core';
+import type {Element} from 'react';
+
+type ExtractReturnType = <V>(() => V) => V;
```

Comment:
I believe this utility might be exported from `fusion-core` however it is also fine to define it here.

Reply:
It's not exported at the top level in core. Would be useful for me to fix it in `fusion-core`?

Reply:
For what it's worth, this is a fairly common pattern in Flow, so it's probably fine to have duplicated.

Reply:
Sounds good